sway-lsp/src/capabilities/hover.rs

```diff
@@ -1,98 +1,108 @@
-                    for document_ref in &session.documents {
-                        if let Some(declared_token) = document_ref.get_declared_token(&token.name) {
-                            if declared_token.is_same_type(token) {
-                                return Some(get_hover_format(declared_token, &session.documents));
-                            }
-                        }
+            if let Some((_, token)) = document.get_token_at_position(position) {
+                if let Some(decl_ident) = document.get_declared_token_ident(token) {
+                    if let Some(decl_token) =
+                        document.get_token_map().get(&to_ident_key(&decl_ident))
+                    {
+                        let hover = get_hover_format(decl_token, &decl_ident);
+                        return Some(hover);
                     }
-                    None
                 }
-            } else {
-                None
             }
+            None
         }
         _ => None,
     }
 }
 
-fn get_hover_format(token: &Token, documents: &Documents) -> Hover {
-    let value = match &token.token_type {
-        TokenType::VariableDeclaration(var_details) => {
-            let var_type = match &var_details.var_body {
-                VarBody::FunctionCall(fn_name) => get_var_type_from_fn(fn_name, documents),
-                VarBody::Type(var_type) => var_type.clone(),
-                _ => "".into(),
-            };
+fn get_hover_format(token: &TokenType, ident: &Ident) -> Hover {
+    let token_name: String = ident.as_str().into();
+    let range = get_range_from_span(&ident.span());
 
-            format!(
-                "let{} {}: {}",
-                if var_details.is_mutable { " mut" } else { "" },
-                token.name,
-                var_type
-            )
-        }
-        TokenType::FunctionDeclaration(func_details) => func_details.signature.clone(),
-        TokenType::StructDeclaration(struct_details) => format!(
-            "{}struct {}",
-            extract_visibility(&struct_details.visibility),
-            &token.name
-        ),
-        TokenType::TraitDeclaration(trait_details) => format!(
-            "{}trait {}",
-            extract_visibility(&trait_details.visibility),
-            &token.name
-        ),
-        TokenType::EnumDeclaration(enum_details) => format!(
-            "{}enum {}",
-            extract_visibility(&enum_details.visibility),
-            &token.name
-        ),
-        _ => token.name.clone(),
+    let format_visibility_hover = |visibility: Visibility, decl_name: &str| -> String {
+        format!(
+            "{}{} {}",
+            extract_visibility(&visibility),
+            decl_name,
+            token_name
+        )
+    };
+
+    let format_variable_hover = |is_mutable: bool, type_name: String| -> String {
+        format!("let{} {}: {}", is_mutable, token_name, type_name,)
```

Comment:
It looks like this would format as something like:

```rust
letfalse token_name: type_name
```

Is that the intent?

Reply:
Nope that's the intent :) good spot!